ang/prospect/services/view-pledge-contribution-case-action.service.js

```diff
@@ -0,0 +1,107 @@
+(function (angular, $, _) {
+  var module = angular.module('prospect');
+
+  module.service('ViewPledgeContributionCaseAction', ViewPledgeContributionCaseAction);
+
+  function ViewPledgeContributionCaseAction ($location, crmApi, ProspectGlobalValues) {
+    var prospectConverted = {
+      isConverted: false,
+      paymentInfo: {
+        payment_completed: false,
+        pledge_balance: false,
+        payment_url: false,
+        payment_entity: false
+      }
+    };
+
+    /**
+     * Refresh Data for the Service
+     *
+     * @param {array} cases
+     */
+    this.refreshData = function (cases) {
+      if (cases[0]) {
+        var caseID = cases[0].id;
+
+        checkIfConvertedToProspect(caseID)
+          .then(function () {
+            getPaymentInfo(caseID);
+          });
+      }
+    };
+
+    /**
+     * Checks if the Action is allowed
+     *
+     * @param {Object} action
+     * @param {Array} cases
+     * @return {boolean}
+     */
+    this.isActionAllowed = function (action, cases) {
+      var isActionAllowed = action.type === 'pledge'
+        ? prospectConverted.paymentInfo.payment_entity === 'pledge'
+        : prospectConverted.paymentInfo.payment_entity === 'contribute';
```

Comment:
isn't this the same:

```js
var isActionAllowed = action.type === prospectConverted.paymentInfo.payment_entity;
```

Reply:
No, because for contribution,
action type is `contribution` and payment_entity is `contribute`. But changed the logic to make it easier to understand

```js
var actionTypeMapping = {
        pledge: 'pledge', contribution: 'contribute'
      };
      var isActionAllowed = actionTypeMapping[action.type] === prospectConverted.paymentInfo.payment_entity;
```